### PR TITLE
JAMES-3830 UserQuotaRootResolver support method list all QuotaRoot of delegated mailboxes

### PR DESCRIPTION
**Pull request by vttranlina**
jira https://issues.apache.org/jira/browse/JAMES-3830
resolve: https://github.com/linagora/james-project/issues/4631

### CODE REVIEW

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/DefaultUserQuotaRootResolver.java [hunk 1]
```diff
@@ -127,6 +128,18 @@ public Publisher<QuotaRoot> getQuotaRootReactive(MailboxPath mailboxPath) {
         return Mono.just(getQuotaRoot(mailboxPath));
     }
 
+    @Override
+    public Publisher<QuotaRoot> listAllAccessibleQuotaRoots(Username username) {
+        MailboxSession session = sessionProvider.createSystemSession(username);
+
+        Flux<QuotaRoot> quotaRootListFromDelegatedMailboxes = factory.getMailboxMapper(session)
+            .findNonPersonalMailboxes(username, MailboxACL.Right.Read)
+            .flatMap(this::getQuotaRootReactive)
```
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/DefaultUserQuotaRootResolver.java:137]:
```suggestion
            .flatMap(this::getQuotaRootReactive, ReactorUtils.DEFAULT_CONCURRENCY)
```

**Review by chibenwa — APPROVED** [review]

**Comment by quantranhong1999** [comment]
```
Failed
org.apache.james.jmap.rfc8621.memory.MemoryQuotaGetMethodTest.quotaGetShouldReturnQuotaRootOfDelegatedMailboxWhenExtension{GuiceJamesServer}
```
related?

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/QuotaGetMethodContract.scala [hunk 2]
```diff
@@ -903,4 +904,297 @@ trait QuotaGetMethodContract {
...
+         |                            "Mail"
+         |                        ],
+         |                        "limit": 100,
+         |                        "warnLimit": 90,
+         |                        "resourceType": "count",
+         |                        "scope": "account"
+         |                    },
+         |                    {
+         |                        "used": 0,
+         |                        "name": "#private&andre@domain.tld@domain.tld:account:count:Mail",
```
Comment (quantranhong1999) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/QuotaGetMethodContract.scala:1041]: Q: How does the client know which (delegated) mailbox this quota applies to? Should we need some kind of associated mailbox id or client just parse this quota name to get the mailbox name?

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/QuotaGetMethodContract.scala:1041]: This is james specific so we could start with "just parse the quota..."

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Comment by vttranlina** [comment]
rebase master

**Comment by quantranhong1999** [comment]
Can we merge this?

**Pull request merged**